scripts/functional-tests.sh

```diff
@@ -124,7 +134,7 @@ cleanup() {
 }
 
 init_test_script
-trap 'cleanup "${?}"; rm -rf "${TEST_DIR}"' EXIT QUIT INT TERM
+trap 'cleanup "${?}"; test -z $CLEANUP_TEST_DIR && rm -rf "${TEST_DIR}"' EXIT QUIT INT TERM
```

Comment:
`-z` means "the length of string is zero". This makes the test match even with `CLEANUP_TEST_DIR=0`, which I don't think is your intent. Suggest to use integer-ish comparison: `test x$CLEANUP_TEST_DIR = x1`.